api/python/t4/packages.py

```diff
@@ -527,48 +546,14 @@ def push(self, name, path):
             A new package that points to the copied objects
         """
         dest = fix_url(path)
-        if dest.startswith('file://'):
-            # local push
-            os.makedirs(path, exist_ok=True)
-            pkg = self._materialize(dest)
-            pkg.build(name)
+        if name:
+            dest = dest.strip('/') + '/' + quote(name) + '/'
+        if dest.startswith('file://') or dest.startswith('s3://'):
+            pkg = self._materialize(dest.strip('/'))
```

Comment:
maybe `dest = fix_url(path).strip('/')` since it looks like you always strip it later